v4/core/base_service_test.go

```diff
@@ -30,12 +32,8 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-type Foo struct {
-	Name *string `json:"name,omitempty"`
-}
-
 // Test a normal JSON-based response.
-func TestGoodResponseJSON(t *testing.T) {
+func TestRequestGoodResponseJSON(t *testing.T) {
```

Comment:
Just modified the names of the functions so that we can use `go test -tags=all -run=TestRequest` to run just the "request"-related tests.